packages/nimble-components/src/table-column/base/index.ts

```diff
@@ -45,5 +48,27 @@ export abstract class TableColumn<
             );
         }
         this.columnInternals = new ColumnInternals(options);
+        this.columnInternals.currentSortDirection = this.sortDirection;
+        this.columnInternals.currentSortIndex = this.sortIndex;
+    }
+
+    protected sortDirectionChanged(): void {
+        this.columnInternals.currentSortDirection = this.sortDirection;
+    }
+
+    protected sortIndexChanged(): void {
+        this.columnInternals.currentSortIndex = this.sortIndex;
+    }
+
+    protected sortingDisabledChanged(): void {
+        if (this.columnInternals) {
+            if (this.sortingDisabled) {
+                this.columnInternals.currentSortDirection = TableColumnSortDirection.none;
+                this.columnInternals.currentSortIndex = undefined;
+            } else {
+                this.columnInternals.currentSortDirection = this.sortDirection;
```

Comment:
The HLD didn't specify what values `currentSortIndex`/`currentSortDirection` should have after `sortingDisabled` transitions from `true` to `false`, but setting them back to `sortDirection`/`sortIndex` seemed reasonable.

Also note that if we didn't do this, we could end up in a weird state where trying to set sortDirection/sortIndex to their previous values would not update currentSortIndex/currentSortDirection, i.e.

```ts
column1.sortIndex = 0;
column1.sortDirection = TableColumnSortDirection.ascending;
// currentSortDirection is now ascending, currentSortIndex is now 0
// ...
column1.sortingDisabled = true;
// currentSortDirection / currentSortIndex are now undefined
// ...
column1.sortingDisabled = false;
column1.sortIndex = 0;
column1.sortDirection = TableColumnSortDirection.ascending;
// the last 2 property sets are no-ops and don't trigger sortIndexChanged() / 
//   sortDirectionChanged() because the properties already had those values
```